momentumopt/srcpy/momentumopt/setting/PySetting.cpp

```diff
@@ -15,6 +16,25 @@
 namespace py = pybind11;
 using namespace momentumopt;
 
+static void setIntParam(PlannerSetting& self,const PlannerIntParam& param, const int value){
+  self.get(param) = value;
+}
+static void setBoolParam(PlannerSetting& self,const PlannerBoolParam& param, const bool value){
+  self.get(param) = value;
+}
+static void setDoubleParam(PlannerSetting& self,const PlannerDoubleParam& param, const double value){
+  self.get(param) = value;
```

Comment:
Does this work? I would assume `self.get(param)` does a copy-by-value and returns the setting parameter as a new double and then I am not sure what the assignment is doing in the end.

Reply:
Yes, I added some assert in the demo script to show this. If self is not const, it will call the method https://github.com/machines-in-motion/kino_dynamic_opt/blob/master/momentumopt/src/momentumopt/setting/PlannerSetting.cpp#L395 which return a non const reference.